db/migrate/20170919120008_move_visibility_to_addresses.rb

```diff
@@ -0,0 +1,35 @@
+class MoveVisibilityToAddresses < ActiveRecord::Migration[5.1]
+  def change
+    reversible do |direction|
+
+      direction.up do
+        add_column :addresses, :visibility, :string, default: 'street_address'
+
+        Company.all.each do |company|
+          # At the time of this migration we have only allowed one address
+          address = company.addresses.first
+          if address
+            address.visibility = company.address_visibility
+            address.save!
+          end
+        end
+
+        remove_column :companies, :address_visibility
+      end
+
+      direction.down do
+        add_column :companies, :address_visibility, :string, default: 'street_address'
+
+        Company.all.each do |company|
+          address = company.addresses.first
+          if address
+            company.address_visibility = address.visibility
+            company.save!
+          end
+        end
+
```

Comment:
If a company has more than one address and we do a rollback, we should be sure to record any companies that will "lose" those addresses (e.g. any addresses > the first one).  I suggest that they be written out to a log & the screen.  Not sure if the log just needs to be the current one (`development` or `test` or `production`) or a specific log file.

Reply:
My thinking on this is that the "rollback" would be valid only while we are still noodling about multiple company addresses, and how best to do that.  Once we actually release production code that allows that, we have crossed the Rubicon and there is not going back (from a data POV). 

So, the rollback here is not hardened in any way for that situation.  In fact, I don't even bother to delete orphaned company address records as a result.  (that is, if we rolled back to having just _one_ company record, any other existing company address records are orphaned (we'll just go back to calling `#main_address` and grabbing the first address record, and ignore the rest)).

Thats why I am not logging information related to orphaned records - we commit to multiple addresses or not, and once we do so we can't really recover the DB records completely nor accurately.

Thoughts?

Reply:
I think that's fine.   

If we had more users  + more data & verified that there were cases in the production data with multiple addresses, well -- then we both could go to town on this.